SpatialGDK/Source/SpatialGDKServices/Private/SpatialGDKServicesModule.cpp

```diff
@@ -24,15 +24,17 @@ DEFINE_LOG_CATEGORY(LogSpatialGDKServices);
 IMPLEMENT_MODULE(FSpatialGDKServicesModule, SpatialGDKServices);
 
 static const FName SpatialOutputLogTabName = FName(TEXT("SpatialOutputLog"));
+TWeakPtr<SSpatialOutputLog> SpatialOutputLog;
```

Comment:
So this can't actually exist in the header at the moment. Including the `SSpatialOutputLog` in the module header causes some cyclic dependency which breaks compilation. This is also why we are forward declaring the class and using `TWeakPtr` rather than a shared reference as then the module header doesn't need to know how to construct the SSpatialOutputLog object. For now this design design will do (it's not invalid it's just not ideal) but could be potentially be improved in the final PR when merging to master.

Reply:
fair enough. I would definitely like to revisit in the final PR, but it might make more sense to review it there to see the whole context.